### ISSUE TITLE
Inconsistent definition of `clip_timestamps` parameter between `WhisperModel` and `BatchedInferencePipeline`

### ISSUE THREAD

**Issue by zh-plus** [SYSTRAN/faster-whisper#1205]
The `clip_timestamps` parameter is defined differently in `WhisperModel` and `BatchedInferencePipeline`, which creates confusion when switching between these classes:

In `WhisperModel.transcribe()`:
https://github.com/SYSTRAN/faster-whisper/blob/1b24f284c9b39bcdf47233d7ae507abaea7a98c8/faster_whisper/transcribe.py#L735

In `BatchedInferencePipeline.transcribe()`:
https://github.com/SYSTRAN/faster-whisper/blob/1b24f284c9b39bcdf47233d7ae507abaea7a98c8/faster_whisper/transcribe.py#L294

**Comment by Purfview** [comment]
Yes, I was thinking to rename it to `vad_timestamps` for batched, but it's "cleaner" as is I think.
Maybe just add the notes at the descriptions that it needs different input in batched vs sequential modes.

**Comment by zh-plus** [comment]
I think unifying the interfaces would be better than documenting differences. We could change both to Optional[List[dict]] where each dict has start/end times. This would make it possible to use `WhisperModel` and `BatchedInferencePipeline` with the same `clip_timestamps` for self-defined VAD.

**Comment by Purfview** [comment]
>We could change both to Optional[List[dict]] 

clip_timestamps in sequential mode is for a quick user input, not for "self-defined VAD", btw, you can't have this functionality in batched mode as audio there is already chunked by VAD.

**Comment by MahmoudAshraf97** [comment]
I had that in mind when naming the parameter, it's not ideal I know, but if I'm going to change anything, then the sequential one is the candidate, because the batched one is much clearer IMO with clear starts and ends
